vendor/src/github.com/docker/go-connections/nat/nat.go

```diff
@@ -146,9 +146,31 @@ func ParsePortSpecs(ports []string) (map[Port]struct{}, map[Port][]PortBinding,
 			rawPort = fmt.Sprintf(":%s", rawPort)
 		}
 
-		parts, err := PartParser(portSpecTemplate, rawPort)
-		if err != nil {
-			return nil, nil, err
+		parts := make(map[string]string)
+		if rawPort != "" && rawPort[0] == '[' {
```

Comment:
I am trying to see if we do really need the square brackets.

The `-p value` can be:
1)  \<Container Port\>
2)  \<Host Port\>:\<Container Port\>
3)  \<Host IP\>:\<Host Port\>:\<Container Port\>

In all the cases, I think, we need to
a) If no `:` are present in value,  then ContainerPort=value, exit
b) else split value by `:` into N elements, ContainerPort=\<Nth element\>, HostPort=\<(N-1)th element\>
     if len(elements) == 3, then parse the IPv4 host address from the first element, exit
     else if len(elements) > 3 join the first N-2 elements in one string and parse the IPv6 from it, exit
     else exit